astropy/io/ascii/tests/test_read.py

```diff
@@ -1047,8 +1047,10 @@ def test_guessing_file_object():
     """
     Test guessing a file object.  Fixes #3013 and similar issue noted in #3019.
     """
-    t = ascii.read(open('data/ipac.dat.bz2', 'rb'))
+    fd = open('data/ipac.dat.bz2', 'rb')
```

Comment:
For all the cases that calls `open`, can we not use context manager so it still exists gracefully if something went wrong with the test itself?

Reply:
Probably a good idea, but I didn't want to change the tests too much.

Reply:
Well, I guess we can always revisit in follow-up PR(s) if you don't want to address this here.

Reply:
I did a few at least. In one, the diff does become substantially larger as the whole test is indented. I decided not to follow through too much on the fits `hdul` tests...